src/checkers/inference/solver/strategy/GraphSolvingStrategy.java

```diff
@@ -71,24 +66,23 @@ public InferenceSolution solve(SolverEnvironment solverEnvironment, Collection<S
                 slots, constraints, lattice);
 
         // Solving.
-        List<Map<Integer, AnnotationMirror>> inferenceSolutionMaps = new LinkedList<Map<Integer, AnnotationMirror>>();
+        List<Pair<Map<Integer, AnnotationMirror>, Collection<Constraint>>> inferenceResults = new LinkedList<>();
 
         if (separatedGraphSolvers.size() > 0) {
             if (solveInParallel) {
                 try {
-                    inferenceSolutionMaps = solveInparallel(separatedGraphSolvers);
+                    inferenceResults = solveInparallel(separatedGraphSolvers);
                 } catch (InterruptedException | ExecutionException e) {
                     e.printStackTrace();
                 }
             } else {
-                inferenceSolutionMaps = solveInSequential(separatedGraphSolvers);
+                inferenceResults = solveInSequential(separatedGraphSolvers);
             }
         }
-       
 
         // Merge solutions.
-        InferenceSolution solution = mergeSolution(inferenceSolutionMaps);
-       
+        InferenceResult solution = mergeInferenceResults(inferenceResults);
```

Comment:
update variable names and comments throughout

Reply:
Done.